airsonic-main/src/main/java/org/airsonic/player/service/PodcastService.java

```diff
@@ -657,10 +657,11 @@ private synchronized File getFile(PodcastChannel channel, PodcastEpisode episode
 
         File channelDir = getChannelDirectory(channel);
 
-        String filename = StringUtil.getUrlFile(episode.getUrl());
+        String filename = channel.getTitle() + " - " + episode.getPublishDate().toString() + " - " + episode.getTitle();
```

Comment:
I know it wasn't there before, but while we're at it, I think we should make sure that `episode.getTitle()` doesn't return something ridiculously long. Windows systems still have a lot of issues with long paths, so I can definitely see it being an issue here.

Reply:
Yeah, good point!

My suggestion would be to trim it once it's concatenated, so that, for the most part, only the episode title gets cut off if it's too long.

Reply:
Sounds good to me!